### PR TITLE
chore(discover): Removes optional useEvent from DiscoverQuery since it should always be set to true now

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
eventsv2 is no longer usable and no longer being used so this prop can now be removed

### CODE REVIEW

File: static/app/utils/discover/discoverQuery.spec.jsx [hunk 1]
```diff
@@ -25,9 +25,9 @@ describe('DiscoverQuery', function () {
   describe('with eventsv2', function () {
```
Comment (narsaynorath) [inline static/app/utils/discover/discoverQuery.spec.jsx:25]: This is still referencing `eventsv2`, is there already an `events` test block here with duplicate coverage and thus we can delete this?

File: static/app/utils/discover/discoverQuery.tsx [hunk 2]
```diff
@@ -56,16 +55,14 @@ function shouldRefetchData(
...
-  const afterFetch = props.useEvents
-    ? (data, _) => {
-        const {fields, ...otherMeta} = data.meta ?? {};
-        return {
-          ...data,
-          meta: {...fields, ...otherMeta},
-        };
-      }
-    : undefined;
+  const endpoint = 'events';
```
Comment (narsaynorath) [inline static/app/utils/discover/discoverQuery.tsx:58]: Maybe we should just hardcode this on line 68 and skip the variable

File: static/app/utils/discover/discoverQuery.tsx [hunk 3]
```diff
@@ -77,16 +74,14 @@ function DiscoverQuery(props: DiscoverQueryComponentProps) {
...
```
Comment (narsaynorath) [inline static/app/utils/discover/discoverQuery.tsx:77]: Same comment as above

**Review by narsaynorath — APPROVED** [review]
Small comments but looks good to me

**Review by dashed — APPROVED** [review]

**Pull request merged**